app/routes/goal_routes.py

```diff
@@ -0,0 +1,109 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.helpers.helpers import validate_model, send_message
+
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal.from_dict(request_body)
+
+        db.session.add(new_goal)
+        db.session.commit()
+        
+    
+        return {"goal" : new_goal.to_dict()}, 201
+    
+    except KeyError:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+
+@goal_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+    if not goals:
+        return make_response(jsonify([]), 200)
```

Comment:
We can omit lines 32-33. If goals is an empty list your list comprehension won't have any elements to iterate over and return an empty list which gets referenced by `results`